crates/driver/src/domain/competition/auction.rs

```diff
@@ -142,66 +143,56 @@ impl Auction {
                 }
             };
 
-            fn max_fill(order: &competition::Order) -> anyhow::Result<U256> {
-                use {
-                    anyhow::Context,
-                    shared::remaining_amounts::{Order as RemainingOrder, Remaining},
+            fn max_sell(order: &competition::Order) -> Option<U256> {
+                let target = order.target().0;
+                let available = match order.partial {
+                    order::Partial::Yes { available } => available.0,
+                    order::Partial::No => target,
                 };
 
-                let remaining = Remaining::from_order(&RemainingOrder {
-                    kind: match order.side {
-                        order::Side::Buy => model::order::OrderKind::Buy,
-                        order::Side::Sell => model::order::OrderKind::Sell,
-                    },
-                    buy_amount: order.buy.amount.0,
-                    sell_amount: order.sell.amount.0,
-                    fee_amount: order.fee.user.0,
-                    executed_amount: match order.partial {
-                        order::Partial::Yes { executed } => executed.0,
-                        order::Partial::No => 0.into(),
-                    },
-                    partially_fillable: match order.partial {
-                        order::Partial::Yes { .. } => true,
-                        order::Partial::No => false,
-                    },
-                })
-                .context("Remaining::from_order")?;
-                let sell = remaining
-                    .remaining(order.sell.amount.0)
-                    .context("remaining_sell")?;
-                let fee = remaining
-                    .remaining(order.fee.user.0)
-                    .context("remaining_fee")?;
-                sell.checked_add(fee).context("add sell and fee")
-            }
+                let sell = util::math::mul_ratio(order.sell.amount.0, available, target)?;
```

Comment:
My immediate thought was that this would overscale the amounts if `available > order.remainig_amount`.
I see that you adjusted the doc comment on `Partial::Yes` to reflect that this should never happen but should we maybe enforce that in a constructor of `Partial`?

Reply:
I added a `TODO` in #1861 as I think there are more type-level guarantees that we should encode in `Order`:

https://github.com/cowprotocol/services/blob/1fe3b2064af07ee951fc309619fec90d2516dc6c/crates/driver/src/domain/competition/order/mod.rs#L228-L229